components/cdc/src/delegate.rs

```diff
@@ -252,16 +264,25 @@ impl Delegate {
             pending: Some(Pending::default()),
             txn_extra_op,
             failed: false,
+            api_version,
+            has_resolver: false,
         }
     }
 
+    pub fn has_resolver(&self) -> bool {
```

Comment:
The call order of `has_resolver` and `subscribe` can lead different results, so it's easy to cause bugs.
How about `fn subscribe(&self, downstream) -> Result<bool>`? The returned boolean indicates whether to build a resolver or not.